src/main/java/io/cdap/plugin/switchcase/route/RoutingSwitch.java

```diff
@@ -0,0 +1,348 @@
+    private final String nullPort;
+
+    Config(@Nullable String routingField, @Nullable String portSpecification, @Nullable String defaultHandling,
+           @Nullable String defaultPort, @Nullable String nullHandling, @Nullable String nullPort) {
+      this.routingField = routingField;
+      this.portSpecification = portSpecification;
+      this.defaultHandling = defaultHandling == null ? DefaultHandling.ERROR_PORT.value : defaultHandling;
+      this.defaultPort = defaultPort == null ? DEFAULT_PORT_NAME : defaultPort;
+      this.nullHandling = nullHandling == null ? NullHandling.ERROR_PORT.value : nullHandling;
+      this.nullPort = nullPort == null ? DEFAULT_NULL_PORT_NAME : nullPort;
+    }
+
+    private void validate(Schema inputSchema, FailureCollector collector) throws IllegalArgumentException {
+      // TODO: Add validation code for mode after adding jexl mode
+      Optional<DefaultHandling> handling = DefaultHandling.fromValue(defaultHandling);
+      if (!handling.isPresent()) {
+        collector.addFailure(String.format("Unsupported default handling value %s", DEFAULT_HANDLING_PROPERTY_NAME),
+                             String.format("Please specify one of %s", Joiner.on(",").join(DefaultHandling.values())))
+          .withConfigProperty(DEFAULT_HANDLING_PROPERTY_NAME);
+      }
+
+      // TODO: Add validation for jexl mode after adding jexl support
+
+      if (routingField == null || routingField.isEmpty()) {
+        collector.addFailure("Field to split is required when routing mode is basic.",
+                             "Please provide the field to split on").withConfigProperty(ROUTING_FIELD_PROPERTY_NAME);
+      }
+      Schema.Field field = inputSchema.getField(routingField);
+      if (field == null) {
+        collector.addFailure(String.format("Routing field %s not found in the input schema", routingField),
+                             "Please provide a field that exists in the input schema")
+          .withConfigProperty(ROUTING_FIELD_PROPERTY_NAME);
+        throw collector.getOrThrowException();
+      }
+      Schema fieldSchema = field.getSchema();
+      Schema.Type type = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
+      if (!ALLOWED_TYPES.contains(type)) {
+        collector.addFailure(
+          String.format("Field to split must be one of - STRING, INTEGER, LONG, FLOAT, DOUBLE, BOOLEAN. " +
+                          "Found '%s'", fieldSchema), null).withConfigProperty(ROUTING_FIELD_PROPERTY_NAME);
+      }
+      if (portSpecification == null || portSpecification.isEmpty()) {
+        collector.addFailure("No port specifications defined.", "Please provide at least 1 port specification")
+          .withConfigProperty(BASIC_PORT_SPECIFICATION_PROPERTY_NAME);
+      }
+    }
+
+    private PortSpecificationEvaluator getPortSpecificationEvaluator(FailureCollector collector) {
+      // TODO: Add code for switching evaluators based on config
```

Comment:
please file a jira.

Reply:
will add right after this PR. have the code ready. just splitting it in multiple PRs. This is a temporary placeholder.